packages/solid/src/reactive/signal.ts

```diff
@@ -1327,7 +1325,7 @@ function runComputation(node: Computation<any>, value: any, time: number) {
   try {
     nextValue = node.fn(value);
   } catch (err) {
-    if (node.pure) Transition && Transition.running ? (node.tState = STALE) : (node.state = STALE);
+    if (node.pure) node[(Transition && Transition.running) ? 'tState' : 'state'] = STALE;
```

Comment:
Most of Transition checks are setup in a way that bundlers like rollup do dead code elimination if people don't use the feature. So we need to verify these still work.  A lot of the conditionals that opt into features, things like dev mode are laid out in a way to consider what the code looks like when the branches are completely removed by the bundler.

I don't know that string array accessors vs property accessors have much performance overhead since engine optimizations are pretty good. But we probably should investigate if we want to do this sort of change.